### PR TITLE
Unify blob size limit breach logging and metering

### PR DESCRIPTION
**Pull request by 3vilhamster**
**What changed?**
I introduced a new log and metric for the blob size limit breach.

**Why?**
We want to track blob size violations for each domain and be able to identify all failed workflows.

**How did you test it?**
Unit tests.

**Potential risks**
For misconfigured domains we can introduce a lot of logs, but that is not expected state.

**Release notes**
Introduced new metric for blob size violations:
blob_size_exceed_limit that can point to issues for a specific domain and operation.

**Documentation Changes**

### CODE REVIEW

File: common/util.go [hunk 1]
```diff
@@ -606,18 +607,32 @@ func CheckEventBlobSizeLimit(
 
 	scope.RecordTimer(metrics.EventBlobSize, time.Duration(actualSize))
 
-	if actualSize > warnLimit {
+	if errorLimit < warnLimit {
 		if logger != nil {
```
Comment (Groxx) [inline common/util.go:611]: is there a reason we wouldn't have a logger?  seems kinda unlikely / something we should just fix...

File: service/frontend/api/handler.go [hunk 2]
```diff
@@ -1174,6 +1178,7 @@ func (wh *WorkflowHandler) RespondActivityTaskFailed(
 		sizeLimitWarn,
 		sizeLimitError,
 		taskToken.DomainID,
+		domainName,
```
Comment (Groxx) [inline service/frontend/api/handler.go:1181]: really makes me wish we had a common name-and-id pair :|  we have access to both almost everywhere...

File: common/util_test.go [hunk 3]
```diff
@@ -20,7 +20,7 @@
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 // SOFTWARE.
 
-package common
+package common_test
```
Comment (Groxx) [inline common/util_test.go:23]: I'm generally in favor of external tests, so this is not a complaint, but: is there a motivation reason for this change?

with the copy/pasted hardcoded values, it feels like a bit of an odd separation.  though I don't think those are useful tests anyway.

File: common/util_test.go [hunk 4]
```diff
@@ -896,79 +935,79 @@ func TestGenerateRandomString(t *testing.T) {
...
 	})
 }
 
 func TestCreateChildContext(t *testing.T) {
 	t.Run("nil parent", func(t *testing.T) {
-		gotCtx, gotFunc := CreateChildContext(nil, 0)
+		gotCtx, gotFunc := common.CreateChildContext(nil, 0)
 		require.Nil(t, gotCtx)
-		require.Equal(t, funcName(emptyCancelFunc), funcName(gotFunc))
+		require.NotNil(t, gotFunc)
```
Comment (Groxx) [inline common/util_test.go:977]: arguably a better test would be to just call it - it's important that it's a no-op func that won't break something.

but yea, not-nil is I think sufficient here if you want to leave it.  it's clear enough.

File: common/util_test.go [hunk 5]
```diff
@@ -1628,13 +1665,74 @@ func TestSecondsToDuration(t *testing.T) {
...
+			blobSize: 21,
+			warnSize: 30,
+			errSize:  20,
+			wantErr:  common.ErrBlobSizeExceedsLimit,
+		},
+	} {
+		t.Run(name, func(t *testing.T) {
+			testScope := tally.NewTestScope("test", nil)
+			metricsClient := metrics.NewClient(testScope, metrics.History)
+			logger := testlogger.New(t)
```
Comment (Groxx) [inline common/util_test.go:1714]: there are simpler noop versions of these, may be nicer if they're not going to be asserted on.

**Review by Groxx — APPROVED** [review]
Some small Qs and nits but overall LGTM.  Seems fine to merge as-is, everything's optional and only if you agree :)

File: common/util_test.go (hunk 3 above)
Comment (3vilhamster) [reply common/util_test.go:23]: The motivation is circular dependencies with our common package, dynamic config, and logger, which prevents us from using it.

Comment (3vilhamster) [reply common/util_test.go:23]: I'll revert this change. I can go with log.NewNoop(). I'll try to craft a separate PR that separates throttledLogger from loggerImpl to break the cycle.

File: common/util.go (hunk 1 above)
Comment (3vilhamster) [reply common/util.go]: Good point. I'm not sure why this existed before. I'll remove it.

File: service/frontend/api/handler.go (hunk 2 above)
Comment (3vilhamster) [reply service/frontend/api/handler.go:1181]: Yep. We do have a domain cache, but it is not propagated everywhere. I f we used DI, we could separate this component and make sure that domain cache is provided, but we don't for now.

File: common/util_test.go (hunk 4 above)
Comment (3vilhamster) [reply common/util_test.go]: Reverted for now, but this is a pointless assertion.

File: common/util_test.go (hunk 5 above)
Comment (3vilhamster) [reply common/util_test.go]: Switched to log.MockLogger, though I'm confused as to why it is not generated.
I guess I'll update the mock in a separate PR. I think mock is manually generated with mockery and has a weird interface for On("Method," args...), which IMHO is even worse than what mock provides.

**Pull request merged**